Following the recent changes from MathComp

pi8027 opened the pull request:
Finmap is broken in the CI of MathComp. This PR attempts to fix the compilation of finmap with the master branches of Coq and MathComp by removing order.v and fixing inheritance of `semisetType` and `setType`. However, there is a `[fset ...]` notation defined twice (with inconsistent level modifiers?) in `finmap.v` and then it does not compile: https://github.com/math-comp/finmap/blob/48c1330c43194c566410bb1dcb1af623b679cc2e/finmap.v#L690-L693 https://github.com/math-comp/finmap/blob/48c1330c43194c566410bb1dcb1af623b679cc2e/finmap.v#L831-L833
```
File "./finmap.v", line 831, characters 0-154:
Error: Notation "[ fset[ _ ] _ | _ : _ , _ : _ ]" is already defined
at level 0 with arguments constr, constr at level 99, constr at level 99,
constr at level 200, constr at level 99, constr
while it is now required to be at level 0 with arguments constr, constr
at level 99, constr at level 99, constr, constr at level 99, constr.
```

@CohenCyril Since I don't see what is the proper fix, could you look at this issue?

ejgallego commented:
Thanks for this @pi8027 , indeed I just hit this problem today. Note that the packaging situation may become a bit tricky as for example if you try install packages in the following way [using 1.11 for current dev]:

- ssreflect 1.11
- finmap 1.10
- algebra 1.11

then the compilation of `ssrint` will fail as `From mathcomp Require Import order.` will pick the stale version in `finmap`.

pi8027 commented:
I think that I initially mixed two different issues. The above notation issue can be reproduced using the master branch of Coq and MathComp [d3f5e11](https://github.com/math-comp/math-comp/commit/d3f5e11aa1bbdf6ee4a111bf4641d162d289340f) without exposing the issue on set.v which this PR fixes. Also, removing L831-L833 of finmap.v makes it compilable.

pi8027 commented:
I thought there is a way to leave order.v in this repository to keep it compatible with the existing official releases of MathComp, but according to the comment by @ejgallego, maybe we should remove it anyway.

CohenCyril commented:
> @CohenCyril Since I don't see what is the proper fix, could you look at this issue?

Hi! I guess the proper fix is to have "Reserved Notation" and I think the second declaration (lines 831-833) is good enough.

pi8027 commented:
Sure. I will do that.

pi8027 commented:
I fixed that notation in #63 and also will redo this PR without removing order.v just to be conservative. Maybe we will remove that later.

pi8027 commented:
Done. I ported recent changes of `order.v` in MathComp using `git-format-patch` and `git-am`, changed `set.v` to make it compatible with non-distributive lattice structures, and reverted some changes to make it compatible with MathComp 1.8.0.
